M2/Macaulay2/m2/Core.m2

```diff
@@ -116,6 +116,21 @@ needs = filename -> if not filesLoaded#?filename then load filename else (
      (filepath, filetime) := filesLoaded#filename;
      if filetime < fileTime filepath then load filepath)
 
+-----------------------------------------------------------------------------
+-- Setup persistent history
+-----------------------------------------------------------------------------
+historyFilename = "~/.Macaulay2-history"
```

Comment:
Perhaps this could be configurable in some way?  For example, Python 3.13 adds support for an environment variable (`PYTHON_HISTORY`) that users can use to set the path of Python's equivalent history file.

Reply:
Ehh, we have a lot of other things that aren't configurable, I don't see the point of starting with this one. But maybe `.Macaulay2/history.m2` would make more sense.

Reply:
That's true.  I do like the idea of putting it in `applicationDirectory()`, since that is configurable by changing `applicationDirectorySuffix`.